onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder_helpers.cc

```diff
@@ -34,8 +34,22 @@
   uint32_t perm_idx = operand_indices.at(perm_input);
 
   input_indices.push_back(perm_idx);  // permutation
-  ORT_RETURN_IF_ERROR(shaper.Transpose(data_input, perm, output));
+
+  auto calculate_transpose_output_shape = [&](const std::string& data_input,
+                                              const std::string& output, const std::vector<int32_t>& perm) -> Status {
```

Comment:
>const std::vector<int32_t>& perm [](http://example.com/codeflow?start=73&length=32)

Conding standards mandate use of gsl::span<const T> to make sure a wide variety of containers is accepted.